File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegateBaseImpl.java
package me.ykrank.s1next.view.internal;

import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.github.ykrank.androidtools.ui.internal.LoadingViewModelBindingDelegate;
import com.github.ykrank.androidtools.ui.vm.LoadingViewModel;

import me.ykrank.s1next.databinding.FragmentBaseBinding;

public final class LoadingViewModelBindingDelegateBaseImpl
        implements LoadingViewModelBindingDelegate {

    private final FragmentBaseBinding binding;

    public LoadingViewModelBindingDelegateBaseImpl(FragmentBaseBinding binding) {
        this.binding = binding;
    }

    @Override
    public View getRootView() {
        return binding.getRoot();
    }

    @Override
    public SwipeRefreshLayout getSwipeRefreshLayout() {
        return binding.swipeRefreshLayout;
    }

    @Override
    public RecyclerView getRecyclerView() {
        return binding.recyclerView;
    }

    @Override
    public void setLoadingViewModel(LoadingViewModel loadingViewModel) {
        binding.setLoadingViewModel(loadingViewModel);
    }
}

File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegateBaseImpl.kt
package me.ykrank.s1next.view.internal

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.github.ykrank.androidtools.ui.internal.LoadingViewModelBindingDelegate
import com.github.ykrank.androidtools.ui.vm.LoadingViewModel
import me.ykrank.s1next.databinding.FragmentBaseBinding

class LoadingViewModelBindingDelegateBaseImpl(private val binding: FragmentBaseBinding) :
    LoadingViewModelBindingDelegate {
    override val rootView: View
        get() = binding.root
    override val swipeRefreshLayout: SwipeRefreshLayout
        get() = binding.swipeRefreshLayout
    override val recyclerView: RecyclerView
        get() = binding.recyclerView

    override fun setLoadingViewModel(loadingViewModel: LoadingViewModel) {
        binding.setLoadingViewModel(loadingViewModel)
    }
}
